crates/cargo-contract/src/crate_metadata.rs

```diff
@@ -139,6 +140,43 @@ impl CrateMetadata {
     }
 }
 
+/// Get the members of a cargo workspace
+pub fn get_cargo_workspace_members(
+    manifest_path: &ManifestPath,
+) -> Result<Vec<PackageId>> {
+    tracing::debug!(
+        "Fetching cargo workspace members for {}",
+        manifest_path.as_ref().to_string_lossy()
+    );
+
+    let mut cmd = MetadataCommand::new();
+    let metadata = cmd
+        .manifest_path(manifest_path.as_ref())
+        .exec()
+        .context("Error invoking `cargo metadata`")?;
+
+    Ok(metadata.workspace_members)
+}
+
+/// Check if a manifest is virtual (dependency graph has no root)
+pub fn is_virtual_manifest(manifest_path: &ManifestPath) -> Result<bool> {
+    let mut cmd = MetadataCommand::new();
+    let metadata = cmd
+        .manifest_path(manifest_path.as_ref())
+        .exec()
+        .context("Error invoking `cargo metadata`")?;
```

Comment:
Can you be more specific here with what went wrong/any suggestions for the user?

Reply:
hmm all I could think was to change the message to something like:
```
    let metadata = cmd
        .manifest_path(manifest_path.as_ref())
        .exec()
        .context(format!(
            "Error invoking `cargo metadata` for {:?} while checking whether workspace is virtual",
            manifest_path
        ))?;
```

would that be enough, or do you mean something more specific?

I copied this code snippet from other parts of the code where it already existed in this form. e.g.:
https://github.com/paritytech/cargo-contract/blob/127ccc952703d322521b5af6d06902748d729d24/crates/cargo-contract/src/crate_metadata.rs#L149-L152

Reply:
Yeah I think your suggestion is better. It at least gives the user a keyword/phrase which they can then use to try and solve their issue